### ISSUE TITLE
Allow to start navigation from current position before having a fix

### ISSUE THREAD

**Issue by jonnius** [rinigus/pure-maps#546]
On some UT devices getting a GPS fix can take up to a few minutes. That means one cannot start navigation.

I have seen users struggling with and asking about it. My personal workaround is to select some random start address to start navigation so I that I can start driving. When it gets a fix, the route is recalculated anyways.

Could we just allow to start navigation from "current position" without having a fix yet, so that the random start address workaround is not needed?

**Comment by rinigus** [comment]
Sure, will look into it.

**Comment by rinigus** [comment]
There isn't much what is required to start navigation. I wonder whether 708b35b827e370d2ed45c6f9362c05eb81df2a79 has helped on UT.

In particular, all what is required that the positioning data is given to Qt. On SFOS it is usually on start of the app. Pure Maps doesn't make any assumptions regarding accuracy while calculating the route. While navigating, it is requested to be accurate, but that makes sense.

Let's see if my change helped - let me know when you get to test it.

**Comment by jonnius** [comment]
I just tested the current `master` and the issues is still the same. The system has GPS enabled and is searching for a fix (an indicator in the top bar shows that). I guess that during that time, Qt does not provide position data. My request here is that Pure Maps allows me to start navigation anyway.

I don't see how starting navigation from "current position" before having a fix (1) is different from starting navigation from an address I enter (2). Navigation will try to locate me and recalculate the route anyway. Pure Maps allows (2) while it does allow (1). The button is disabled until having a fix or entering a random start address.

**Comment by jonnius** [comment]
If you think, the system should behave differently, I would open an issue in the Ubuntu Touch issue tracker instead. In that case, could you describe what you would expect from it?

**Comment by rinigus** [comment]
But is there such a location as a current location displayed by Pure Maps? Does it show some coordinates as a blue dot on the map?

**Comment by jonnius** [comment]
Nope. There is no current location shown.

**Comment by rinigus** [comment]
But in that case Qt does not even provide any kind of valid location estimate. Not much we can do on Pure Maps side.

**Comment by jonnius** [comment]
Why can't we switch to navigation view where route calculation would start once there is a location? If I add a random start address the same happens.

**Comment by rinigus** [comment]
OK, I missed the whole point of it. So, we would need some kind of mode or ability to preorder navigation mode to some destination(s) before the origin is known. I'll have to think how to do that.

**Comment by jonnius** [comment]
I would expect this to be the same issue. But I'll try and report back.

**Comment by rinigus** [comment]
@jonnius : any news regarding it?

**Comment by jonnius** [comment]
It is still the same issue.

**Comment by rinigus** [comment]
To be clear, this is on UT device that used to have GPS fix before and wasn't reflashed? So, on UT, I open Pure Maps, get the fix, close it, wait a bit of time, and start Pure Maps again. On the second start, there is no estimate such as old fix coordinate available, "center on location" cannot center anywhere. Id that correct?

On start, Pure Maps writes "Acquired QGeoPositionInfoSource:" with the name of the plugin. I wonder what is that plugin in UT.

**Comment by jonnius** [comment]
> Acquired QGeoPositionInfoSource: "ubuntu"

Correct. You don't get a position estimation based on the last fixed. You always need to wait for a new fix.

**Comment by rinigus** [comment]
Do you know why is it so? On SFOS, we have position cache and it is set to the last known location on new start. We also have MLS based fix if you install MLS offline datasets (as in https://openrepos.net/content/blacksheepdev/mls-manager). Sounds like nothing like it on UT.

Do you have a link to QGeoPositionInfoSource implementation of "ubuntu" plugin?

**Comment by jonnius** [comment]
Sorry, I lost track of this one. But it is still relevant. So here is the link you asked for: https://gitlab.com/ubports/development/core/lomiri-location-service/-/blob/main/qt/position/core_geo_position_info_source.cpp
